src/imagery/i.asf.download/i.asf.download.py

```diff
@@ -0,0 +1,213 @@
+# % key: l
+# % description: Only list scenes available
+# %end
+
+# %flag
+# % key: c
+# % description: Do checksum test of all scenes
+# %end
+
+# %rules
+# % required: output_directory,-l
+# % required: start
+# % collective: start,end
+# %end
+
+import zipfile
+import os
+import json
+import hashlib
+import concurrent.futures
+
+from pathlib import Path
+
+import shapely.wkt
+import asf_search as asf
+import geopandas as gpd
+
+import grass.script as gs
+
+def checksum_test(checksum, expected_checksum, _dfile):
+    if checksum != expected_checksum:
+        gs.verbose(f'Checksum test failed for {os.path.split(_dfile)[1]}')
+        os.remove(_dfile)
+        gs.verbose(f'{os.path.split(_dfile)[1]} is deleted')
+        return False
+    else:
+        gs.verbose(f'Checksum test OK for {os.path.split(_dfile)[1]}')
+        return True
+    
+
+def main():
+    aoi = shapely.wkt.dumps(gpd.read_file(options["aoi"]).iloc[0].geometry)
+
+    opts = {
+            'platform': options["platform"],
+            'beamMode': options["beamMode"],
+            'processingLevel': options["processinglevel"],
+            'start': options["start"],
+            'end': options["end"],
+            'intersectsWith': aoi
+        }
+
+    results = asf.geo_search(**opts)
+    gs.verbose(f'{len(results)} results found')
+    
+    gs.verbose(f'Listing all scenes found:')
+    for count, r in enumerate(results, start=1):
+        gs.verbose(f'{r.properties['fileName']}')
+    
+    if flags["l"]:
+        gs.verbose(f'All available scenes are listed.')
+        sys.exit()
+    
+    token_file = Path(options["token"])
+
+    with open(token_file, 'r') as f:
+        token = f.read()
+        print(token)
+        f.close()
+        
+    token_session = asf.ASFSession().auth_with_token(token)
+
+    download_dir = Path(options["output_directory"]).mkdir(exist_ok=True, parents=True)
+    download_dir = Path(options["output_directory"])
+    gs.verbose(f'Downloading to {download_dir}')
+    
+    for count, r in enumerate(results, start=1):
+        _dfile = download_dir / r.properties['fileName']
+        
+        r.download(path=download_dir, session=token_session)  
+        gs.verbose(f'{r.properties['fileName']} is downloaded')
+        
+        if flags['c']:
+            i = 0
+            while  i < 3:
+                expected_checksum = r.properties['md5sum']
+                checksum = hashlib.md5(_dfile.read_bytes()).hexdigest()
+                
+                if not checksum_test(checksum, expected_checksum, _dfile):
+                    gs.verbose(f'Trying to download {os.path.split(_dfile)[1]} again')
+                    r.download(path=download_dir, session=token_session)
+                    gs.verbose(f'{r.properties['fileName']} is downloaded again')
+                    i += 1
+                else:
+                    i = 3
+           
+                
+
+if __name__ == "__main__":
+    options, flags = grass.parser()
+    main()
```

Comment:
If you get the wkt geometry with just GDAL these imports are not needed...
```suggestion
    try:
        import shapely.wkt
   except ImportError:
       gs.fatal(_("Can not import shapely. Please install it with 'pip install shapely'"))

    try:
        import geopandas as gpd
   except ImportError:
       gs.fatal(_("Can not import geopandas. Please install it with 'pip install geopandas'"))

        main()
```